lib/reporters/base.js

```diff
@@ -188,18 +193,21 @@ function stringifyDiffObjs(err) {
  * @param {string} expected
  * @return {string} Diff
  */
+
 var generateDiff = (exports.generateDiff = function (actual, expected) {
   try {
-    const diffSize = 2048;
-    if (actual.length > diffSize) {
-      actual = actual.substring(0, diffSize) + ' ... Lines skipped';
-    }
-    if (expected.length > diffSize) {
-      expected = expected.substring(0, diffSize) + ' ... Lines skipped';
-    }
-    return exports.inlineDiffs
+    const maxLen = exports.maxDiffSize || Number.MAX_SAFE_INTEGER;
+    const skipped = Math.max(actual.length - maxLen, expected.length - maxLen);
+
+    actual = actual.slice(0, maxLen);
+    expected = expected.slice(0, maxLen);
+    let result = exports.inlineDiffs
       ? inlineDiff(actual, expected)
       : unifiedDiff(actual, expected);
+    if (skipped > 0) {
+      result = `${result}\nNOTE: Output truncated to ${maxLen} characters, see "maxDiffSize" reporter option.`;
+    }
+    return result;
```

Comment:
Your solution is nice, but please leave the `... Lines skipped` part as is. Changing the output would be a breaking change, I don't want this for now.

Reply:
Sure thing, fixed this!

Reply:
I'm not sure about this one, the output will still be different. We will see when doing some final tests ...

Reply:
The current implementation is broken in my opinion. The "... Lines skipped" part will not be displayed to the user if the diffing part of the string is after the first 2048 bytes (expected and actual will be equal then).

Reply:
Yes, you are right.

Reply:
I did some testing, the message doesn't look nice. Since current solution is broken, we have to change the output anyway.
So your first proposition is better.
`${result}\n......[mocha] output truncated to ${maxLen} characters, see "maxDiffSize" reporter-option\n`;
You have to indent the line with six(?) spaces.

Reply:
Ok, will fix!

Reply:
Done